packages/loot-core/src/server/budget/template-notes.ts

```diff
@@ -0,0 +1,117 @@
+import { Notification } from '../../client/state-types/notifications';
+import * as db from '../db';
+
+import { parse } from './goal-template.pegjs';
+import {
+  CategoryWithTemplateNote,
+  getActiveSchedules,
+  getCategoriesWithTemplateNotes,
+  resetCategoryGoalDefsWithNoTemplates,
+} from './statements';
+import { Template } from './types/templates';
+
+export const TEMPLATE_PREFIX = '#template';
+export const GOAL_PREFIX = '#goal';
+
+export async function storeTemplates(): Promise<void> {
+  const categoriesWithTemplates = await getCategoriesWithTemplates();
+
+  for (const { id, templates } of categoriesWithTemplates) {
+    const goalDefs = JSON.stringify(templates);
+
+    await db.update('categories', {
+      id,
+      goal_def: goalDefs,
+    });
+  }
+
+  await resetCategoryGoalDefsWithNoTemplates();
+}
+
+type CategoryWithTemplates = {
+  id: string;
+  name: string;
+  templates: Template[];
+};
+
+export async function checkTemplates(): Promise<Notification> {
+  const categoryWithTemplates = await getCategoriesWithTemplates();
+  const schedules = await getActiveSchedules();
+  const scheduleNames = schedules.map(({ name }) => name);
+  const errors: string[] = [];
+
+  categoryWithTemplates.forEach(({ id, name, templates }) => {
+    templates.forEach(template => {
+      if (template.type === 'error') {
+        errors.push(`${name}: ${template.line}`);
+      } else if (
+        template.type === 'schedule' &&
+        !scheduleNames.includes(template.name)
+      ) {
+        errors.push(`${id}: Schedule “${template.name}” does not exist`);
+      }
+    });
+  });
+
+  if (errors.length) {
+    return {
+      sticky: true,
+      message: 'There were errors interpreting some templates:',
+      pre: errors.join('\n\n'),
+    };
+  }
+
+  return {
+    type: 'message',
+    message: 'All templates passed! 🎉',
+  };
+}
+
+async function getCategoriesWithTemplates(): Promise<CategoryWithTemplates[]> {
```

Comment:
This no longer takes in a category.  Was that not utilized anywhere?

Reply:
In the original function, the category parameter was always passed in as null, so as far as I could see it was never used